src/main/java/com/cloudbees/jenkins/plugins/bitbucket/api/BitbucketBuildStatus.java

```diff
@@ -72,7 +72,7 @@ public BitbucketBuildStatus(String hash, String description, String state, Strin
         this.description = description;
         this.state = state;
         this.url = url;
-        this.key = key;
+        this.key = key.substring(0, Math.min(key.length(), 40));
```

Comment:
This could cause bizarre results when a shortened key collides with another (likely possible).
IMO A message should be logged in the build console to warn the user (only in the case key.length > 40)